### PR TITLE
Add default load path unconditionially

### PR DESCRIPTION
**Pull request by nuclearsandwich**
Adds /opt/ros/rosdistro/lib to default library path.

Previously this library path was added to LD_LIBRARY_PATH by environment hooks in other packages. However this means that pure-CMake packages which don't depend on ament libraries and do not set environment hooks of their own aren't found by dependents which are also pure cmake.

An example of this is fastrtps and fastcdr, both of which are third-party cmake packages which do not register environment hooks.

This bug was previously masked for fastrtps and fastcdr but a mis-applied dependency on ament_cmake rather than cmake in fastrtps which provided the necessary environment hook to ros_workspace.

### CODE REVIEW

**Comment by dirk-thomas** [comment]
For Bouncy that is probably the quickest way to address the problem. So :+1: to merge and release this for now.

Please add a note to this patch though that this should be revisited in the future. First it was done for arch specific lib directories, now for the general `lib` directory and using the same rational `bin` should also be on the `PATH`.

**Comment by nuclearsandwich** [comment]
> using the same rationale bin should also be on the PATH

Do we have any third party packages which add binaries? If so, think adding the bin directory to the path is something we should do now rather than go through another rebuild cycle.

**Comment by dirk-thomas** [comment]
> Do we have any third party packages which add binaries?

Not that I am aware of.

**Comment by mikaelarguedas** [comment]
> Please add a note to this patch though that this should be revisited in the future.

:+1:

> > using the same rationale bin should also be on the PATH

> Do we have any third party packages which add binaries? If so, think adding the bin directory to the path is something we should do now rather than go through another rebuild cycle.

I don;t think so. The only type of binaries provided by third parties I can think of would be the implementation specific provided code generators. As we use introspection for Fast-RTPS and that the other implementations provide their own binaries I think we're good.

File: CMakeLists.txt [hunk 1]
```diff
@@ -3,15 +3,24 @@ project(ros_workspace)
 
 find_package(ament_cmake_core REQUIRED)
 
+
+
```
Comment (dirk-thomas) [inline CMakeLists.txt]: Please remove the unrelated added newlines.

**Review by dirk-thomas — APPROVED** [review]
LGTM (beside the nitpick comment).

**Comment by nuclearsandwich** [comment]
I've created an issue stub #7 which is now linked in d97f698. I'll add context to that issue for the post release discussion during some downtime later this week. Since it doesn't seem like we expect there to be third party binaries I won't add /opt/ros/rosdistro/bin to the path in a default environment hook.

Edit rebased out stray newlines and updated above commit id.

**Comment by nuclearsandwich** [comment]
As expected, this did resolve the issue building fastrtps without ament_cmake.

**Pull request merged**